community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/checkpoint/CheckPointScheduler.java

```diff
@@ -78,12 +103,16 @@ public boolean getAsBoolean()
             return !checkPointing;
         }
     };
+    private volatile Throwable[] failures = new Throwable[MAX_CONSECUTIVE_FAILURES_TOLERANCE];
```

Comment:
On the topic of `failures` and `consecutiveFailures` this check point job is run by the job scheduler and so can potentially be run by a different thread each time. It's not designed to be run concurrently, nor is it run concurrently in practice. I set the `failures` to `volatile` because this task may have been created after a scheduler thread set to run it in the future was created. Although you may be right that `consecutiveFailures` guards for that with its volatility. Then again, this is a task that is run once every couple of minutes so micro performance concerns aren't of uttermost importance.

On the topic of clearing `failures`, I'd have thought that leaving those exceptions there would not hurt, but you're right it's unnecessary. I'll make that change.